cmd/cluster-agent/api/agent/agent.go

```diff
@@ -53,16 +53,8 @@ func SetupHandlers(r *mux.Router, wmeta workloadmeta.Component, ac autodiscovery
 	}).Methods("GET")
 	r.HandleFunc("/config", settings.GetFullConfig("")).Methods("GET")
 	r.HandleFunc("/config/list-runtime", settings.ListConfigurable).Methods("GET")
-	r.HandleFunc("/config/{setting}", func(w http.ResponseWriter, r *http.Request) {
-		vars := mux.Vars(r)
-		setting := vars["setting"]
-		settings.GetValue(setting, w, r)
-	}).Methods("GET")
-	r.HandleFunc("/config/{setting}", func(w http.ResponseWriter, r *http.Request) {
-		vars := mux.Vars(r)
-		setting := vars["setting"]
-		settings.SetValue(setting, w, r)
-	}).Methods("POST")
+	r.HandleFunc("/config/{setting}", settings.GetValue).Methods("GET")
+	r.HandleFunc("/config/{setting}", settings.SetValue).Methods("POST")
```

Comment:
You are registering the routes twice if I'm not mistaken. 

You are providing via FX value groups here: 
https://github.com/DataDog/datadog-agent/blob/72800a58f5c5f3c0179d23c5d614fc9da65d882c/comp/core/settings/settingsimpl/settingsimpl.go#L226-L229

And registering in lines 54-57. Is that intended?

Reply:
to my knowledge (could be wrong) but the cluster agent does not use the api component so wouldn't use the value groups, instead of [comp/api/api used in the agent run command](https://github.com/DataDog/datadog-agent/blob/main/cmd/agent/subcommands/run/command.go#L44) the [cluster-agent uses the cmd/cluster-agent/api pkg](https://github.com/DataDog/datadog-agent/blob/main/cmd/cluster-agent/subcommands/start/command.go#L28) so this is just fixing the type signature of the function